### ISSUE TITLE
Create dataset loader for SleukRith Set

### ISSUE THREAD

**Issue by SamuelCahyawijaya** [SEACrowd/seacrowd-datahub#206]
Dataloader name: `sleukrith_ocr/sleukrith_ocr.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?sleukrith_ocr

|     Dataset| sleukrith_ocr |
|-------------|---|
| Description | SleukRith Set is the first dataset specifically created for Khmer palm leaf manuscripts. The dataset consists of annotated data from 657 pages of digitized palm leaf manuscripts which are selected arbitrarily from a large collection of existing and also recently digitized images. The dataset contains three types of data: isolated characters, words, and lines. Each type of data is annotated with the ground truth information which is very useful for evaluating and serving as a training set for common document analysis tasks such as character/text recognition, word/line segmentation, and word spotting. |
| Subsets | - |
| Languages | khm |
| Tasks | Optical Character Recognition |
| License | Unknown (unknown) |
| Homepage | https://github.com/donavaly/SleukRith-Set |
| HF URL | - |
| Paper URL | https://dl.acm.org/doi/10.1145/3151509.3151510 |

**Comment by Gyyz** [comment]
#self-assign

**Comment by Gyyz** [comment]
no idea on what kind of the binary file, tried several package like pickle..., but failed

**Comment by holylovenia** [comment]
> https://github.com/donavaly/SleukRith-Set

Hi @Gyyz, the data homepage specifies this [data format](https://drive.google.com/file/d/1I9iZMpux3DO9CjfZtRyfQNHqT75yKT9D/view). I'll copy it here for convenience.

```
data file
- first 4 bytes (integer): width
- next 4 bytes (integer): height
- next 4 bytes (integer): nb samples
- width*height bytes per image (1 byte per pixel)

label file
- first 4 bytes (integer): nb classes
- next 4 bytes (integer): nb samples
- 4 bytes (integer) per label (label value in [0, nb_class[)
```

**Comment by Gyyz** [comment]
> > https://github.com/donavaly/SleukRith-Set
> 
> Hi @Gyyz, the data homepage specifies this [data format](https://drive.google.com/file/d/1I9iZMpux3DO9CjfZtRyfQNHqT75yKT9D/view). I'll copy it here for convenience.
> 
> ```
> data file
> - first 4 bytes (integer): width
> - next 4 bytes (integer): height
> - next 4 bytes (integer): nb samples
> - width*height bytes per image (1 byte per pixel)
> 
> label file
> - first 4 bytes (integer): nb classes
> - next 4 bytes (integer): nb samples
> - 4 bytes (integer) per label (label value in [0, nb_class[)
> ```

Yes, they specify the data format, but haven't specified the file format

> > https://github.com/donavaly/SleukRith-Set
> 
> Hi @Gyyz, the data homepage specifies this [data format](https://drive.google.com/file/d/1I9iZMpux3DO9CjfZtRyfQNHqT75yKT9D/view). I'll copy it here for convenience.
> 
> ```
> data file
> - first 4 bytes (integer): width
> - next 4 bytes (integer): height
> - next 4 bytes (integer): nb samples
> - width*height bytes per image (1 byte per pixel)
> 
> label file
> - first 4 bytes (integer): nb classes
> - next 4 bytes (integer): nb samples
> - 4 bytes (integer) per label (label value in [0, nb_class[)
> ```



> > https://github.com/donavaly/SleukRith-Set
> 
> Hi @Gyyz, the data homepage specifies this [data format](https://drive.google.com/file/d/1I9iZMpux3DO9CjfZtRyfQNHqT75yKT9D/view). I'll copy it here for convenience.
> 
> ```
> data file
> - first 4 bytes (integer): width
> - next 4 bytes (integer): height
> - next 4 bytes (integer): nb samples
> - width*height bytes per image (1 byte per pixel)
> 
> label file
> - first 4 bytes (integer): nb classes
> - next 4 bytes (integer): nb samples
> - 4 bytes (integer) per label (label value in [0, nb_class[)
> ```

Yes, thanks, they specified the **data format**, but haven't specified the **file format**. No idea on how to load the data.

**Comment by ssun32** [comment]
#self-assign

**Comment by akhdanfadh** [comment]
#self-assign

**Comment by akhdanfadh** [comment]
Hi, I managed to load the data. There are 113206 48*48 images with 111 labels, but these labels are just numbers with no corresponding text. Should I just load it as is?

![Image](https://github.com/SEACrowd/seacrowd-datahub/assets/64322471/df65b3ae-cb8a-47ed-96c4-8b3b44d8f30e)

**Comment by holylovenia** [comment]
Wow, amazing at least you can load the data! Wait, let me come back to this tonight.

**Comment by akhdanfadh** [comment]
Waiting for further instructions before GitHub adding staled-issue @sabilmakbar @holylovenia

**Comment by holylovenia** [comment]
> Waiting for further instructions before GitHub adding staled-issue @sabilmakbar @holylovenia

Sorry I missed this. 🙏 @sabilmakbar, do you have any suggestions on the download method?

Alternatively, if downloading the dataset via the dataloader is too difficult, we can use `_LOCAL = True`. For `_LOCAL = True` dataloaders, usually it's up to the user where to store the data. Later the user can use the `data_dir` flag/parameter to load the data.

**Comment by akhdanfadh** [comment]
> For _LOCAL = True dataloaders, usually it's up to the user where to store the data. Later the user can use the data_dir flag/parameter to load the data.

@holylovenia Got it, answers the question. Probably using `_LOCAL = True` for this dataset.

---

> > My main question here is actually where should I store the downloaded dataset for local data? For context, datasets.DownloadManager stores them in a $HOME/.cache folder, and I can't use **them** for this.

@sabilmakbar What I meant by _them_ here is the `DownloadManager`. I don't know how reliable downloading from Google Drive is using the function, see my result below.
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/64322471/beeadeec-110c-4ded-a44a-f3536fb2b4de)
